components/shared/net/request.rs

```diff
@@ -208,7 +208,11 @@ impl RequestBody {
     }
 
     pub fn len(&self) -> Option<usize> {
-        self.total_bytes.clone()
+        self.total_bytes
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.total_bytes == Some(0)
```

Comment:
I don't know about this `is_empty`. What should the return value be when `total_bytes` is None?

Reply:
You are right, I think it makes sense that it returns false if it is None or if it is Some(0), what do you think?

Reply:
Hrm. I think maybe if the function will ultimately be unused it might better to simply not add it, since it won't have any test coverage and I'm not sure what it should mean in this case. What do you think?

Reply:
I think that makes sense, I can simply add an allow so clippy doesn't trigger the warning with an explanation comment, and if in the future it is needed then it can be implemented.

Reply:
Actually I just saw that #31551 is specifically for this file so its probably best handled there, and I'll just keep this out when I make the new prs